sdks/java/core/src/test/java/org/apache/beam/sdk/util/IOChannelUtilsTest.java

```diff
@@ -80,6 +80,13 @@ public void testLargeShardCount() {
   }
 
   @Test
+  public void testFilePrefix() throws Exception {
+    IOChannelUtils.getFactory("file://tmp");
+    IOChannelUtils.getFactory("file:/tmp");
+    IOChannelUtils.getFactory("file:tmp");
```

Comment:
More specifically, the FileIOChannelFactory `resolve` method (among others, I believe) interprets the URI scheme as a relative path rather than a scheme; i.e. the URI `"file:///foo/bar"` becomes `$PWD/file:/foo/bar` rather than the correct `/foo/bar`